tests/lib/rules/no-const-assign.js

```diff
@@ -30,6 +30,20 @@ ruleTester.run("no-const-assign", rule, {
 		// ignores non constant.
 		"var x = 0; x = 1;",
 		"let x = 0; x = 1;",
+		{
+			code: "using resource = f(); resource = g();",
+			languageOptions: {
+				sourceType: "module",
+				ecmaVersion: 2026,
+			},
+		},
+		{
+			code: "await using resource = f(); resource = g();",
+			languageOptions: {
+				sourceType: "module",
+				ecmaVersion: 2026,
+			},
+		},
```

Comment:
Per https://github.com/eslint/eslint/issues/19792#issuecomment-2922568363, these cases should be `invalid`. I prepared a separate PR for this rule (https://github.com/eslint/eslint/pull/19892) so we can remove these tests.